app/src/main/java/com/futurice/freesound/feature/search/SearchFragmentViewModel.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.feature.search;
+
+import com.futurice.freesound.feature.common.DisplayableItem;
+import com.futurice.freesound.feature.common.Navigator;
+import com.futurice.freesound.network.api.model.Sound;
+import com.futurice.freesound.viewmodel.BaseViewModel;
+
+import android.support.annotation.NonNull;
+
+import java.util.List;
+
+import io.reactivex.Observable;
+import io.reactivex.disposables.CompositeDisposable;
+import polanski.option.Option;
+
+import static com.futurice.freesound.feature.common.DisplayableItem.Type.SOUND;
+import static com.futurice.freesound.utils.Preconditions.get;
+
+final class SearchFragmentViewModel extends BaseViewModel {
+
+    @NonNull
+    private final SearchDataModel searchDataModel;
+
+    @NonNull
+    private final Navigator navigator;
+
+    SearchFragmentViewModel(@NonNull final SearchDataModel searchDataModel,
```

Comment:
Out of consistency, I tend not to use `@Inject`. It's personally preference, but I like being able to visit the module class to see all the dependencies and their scoping in the one place.